### PR TITLE
Refactor/modularize stake pool queries

### PR DESCRIPTION
**Pull request by jvieiro**
# Context

In order to improve testing quality, modularize `DbSyncStakePoolSearch` queries

# Proposed Solution

Create a new dependency into `DbSyncStakePoolSearch` called `StakePoolSearchBuilder` with the previous `DbSyncStakePoolSearch's` private methods (query builders and query calls).

# Important Changes Introduced

n/a

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]
Looks good to me, just need to squash to 1 commit after approvals.
@iadmytro are you verifying the snapshots manually?

**Review by rhyslbw — APPROVED** [review]

File: packages/cardano-services/test/StakePoolSeach/DbSyncStakePoolSearchProvider/StakePoolSearchBuilder.test.ts
Comment (iadmytro) [inline packages/cardano-services/test/StakePoolSeach/DbSyncStakePoolSearchProvider/StakePoolSearchBuilder.test.ts:88]: It would be good to add test for query itself, not only query builder

**Review by iadmytro — CHANGES_REQUESTED** [review]

File: packages/cardano-services/test/StakePoolSeach/DbSyncStakePoolSearchProvider/StakePoolSearchBuilder.test.ts
Comment (jvieiro) [reply packages/cardano-services/test/StakePoolSeach/DbSyncStakePoolSearchProvider/StakePoolSearchBuilder.test.ts:88]: The query itself is being used to build subqueries at `buildOrQuery` and `buildAndQuery`, and then run at queryPoolHashes. So it should be already tested there. Anyway I added a few more validations that includes running the query.

**Review by iadmytro — APPROVED** [review]

**Review by rhyslbw — APPROVED** [review]
Great work @jvieiro

**Pull request merged**
